[Tan Yi Xian] iP

yixiann opened the pull request:
# DukePro
> "Your mind is for having ideas, not holding them." - David Allen
> [(source)](www.google.com)
DukePro frees your mind of having to remember things you need to do. It's, 
- text-based
- easy to learn
- ~FAST~ *SUPER FAST* to use

All you need to do is,
1. Download it from [here](www.google.com).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
add Duke Level-1: Greet, Echo, Exit

Commit:
add Duke Level-2: Add, List

Commit:
add Duke Level-3: Mark as Done

Commit:
add Duke Level-4: ToDo, Event, Deadline

Commit:
add Duke A-TextUiTesting

Commit:
add Duke Level-5: Handle Errors

Commit:
add Duke Level-6: Delete

Commit:
Revert "add Duke Level-6: Delete"

This reverts commit 41609d11c143fd3997cde93f671b6cb6e5ff16a4.

Commit:
Revert "Revert "add Duke Level-6: Delete""

This reverts commit 538c031ed893fb5a927c7631d6b5452a56e86456.

Commit:
add enums and fix exceptions

Commit:
add javadocs

Commit:
Amend switch case tab formatting

Commit:
Change task to abstract class

Commit:
Use string builder to list task instead of system.out.println

Commit:
Add saving functionality to duke

Commit:
Add "/data/dukeData.text" to .gitignore

Commit:
Convert String date in event task and deadline task into localDate

Commit:
Merge conflict

Commit:
Remove "this"

Commit:
Consolidate into Duke package and use OOP.

Commit:
Merge remote-tracking branch 'coworker/add-gradle-support'

Commit:
Add gradle

Commit:
Add JUnit

Commit:
Add ability to create a JAR file

Commit:
Add more javadocs

Commit:
Add standardisation

Commit:
Add find ability

Commit:
Add additional changes to follow coding standard

Commit:
Merge branch 'heads/Level-9'

Commit:
Merge conflicts

florentianayuwono commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,192 @@
+    }
+
+    public static void main(String[] args) {
+        System.out.println("|".repeat(60));
```
Perhaps this function could be handled by the ui?

yixiann replied:
Noted

florentianayuwono commented on src/main/java/Duke/Ui.java:
```diff
@@ -0,0 +1,148 @@
+    /**
+     * Prints a success message when a file is successfully loaded.
+     */
+    public void fileSuccessfullyLoaded() {
```
Perhaps this can be written as `loadFileSuccessfully()` to match the coding standard given below?

> Names representing methods must be verbs and written in camelCase.

yixiann replied:
Thank you for raising this point. Will amend accordingly!

florentianayuwono commented on src/main/java/Duke/enums/Command.java:
```diff
@@ -0,0 +1,53 @@
+ * Command enum represents the possible commands that a user can provide to duke.
+ * The value of these commands are the string the user requires inputting.
+ */
+public enum Command {
```
I like how you you use Enums to represent commands.👍

florentianayuwono commented on src/main/java/Duke/exceptions/FileFailedToLoadException.java:
```diff
@@ -0,0 +1,14 @@
+ * {@code FileFailedToLoadException} is an exception thrown when a
+ * file fails to load.
+ */
+public class FileFailedToLoadException extends DukeException {
```
I like how you create child classes for `DukeException`.

florentianayuwono commented on src/main/java/Duke/utils/Utils.java:
```diff
@@ -0,0 +1,56 @@
+import java.time.format.DateTimeParseException;
+
+/**
+ * The {@code Utils} class contains useful utilities used throughout the application.
```
I like that you have JavaDoc for almost (if not all) every methods and fields.

florentianayuwono reviewed (commented):
Overall I think you did a very good job! The code is very neat and well written. I learn a lot from your implementation here. 🌞

Commit:
Add GUI

Commit:
Rename variables for consistency and apply more SLAP to GUI and storage

Commit:
Replace java fx with java fxml.

Commit:
Add assertions to Duke

There are no assertions throughout the application.

It is recommended that assertions are used liberally in the code. Their impact on performance is considered low and worth the additional safety they provide.

Adding assertions to methods in duke related to task lists and storage allows us to ensure an unusual condition created by the user raises an exception.

In methods that require tasks list, we assert that task list is not null. In methods that require the storage, we assert that the storage is not null. In methods that require at least one task, we assert that the task list is not empty.  The same concept is applied to file and inputs.

yixiann committed:
Merge pull request #2 from yixiann/branch-A-Assertions

Add assertions to Duke

Commit:
Improve coding standards

There are some violations in coding standards throughout the application.

We should practice writing code of high quality.
Given how the world is becoming increasingly dependent on software,
poor quality code is something no one can afford to tolerate.

There are two main improvements made in this commit.
*Avoid Complicated Expressions
*Make the Happy Path Prominent

This is done to increase readability of code which will lead to increase
efficiency in debugging and extension in the future.

Javadocs were added to some methods thoughout the application as well.

yixiann committed:
Merge pull request #3 from yixiann/branch-A-CodeQuality

Improve coding standards

Commit:
Add Help Command

There are no commands to show the user what commands there are.
The only help occurs when the user types a wrong command.

We should be explicit in allowing the user know they can get help in using Duke.
We must implement a help command so users can get help.

Commit:
Improve UI by adding colours, circle profile pictures and drop shadows

Commit:
Add Neko Personality to Duke

Commit:
Add line breaks to some string responses to beautify neko response and add ui.png file to docs

Commit:
Rename package from "Duke" to "duke" to follow package name style and modify headers to adhere to header comments style

Commit:
Improve GUI by allowing a bit of resizing

Commit:
Update ui.png file to show newly added improvements

yixiann committed:
Update user guide

Commit:
Add stage title

Commit:
Merge branch 'master' of https://github.com/yixiann/ip

Commit:
Update ui image to show title of application

yixiann committed:
Update user guide to escape [x] in markdown

Commit:
Rename ui to Ui for image of user interface

Commit:
Reorder examples to be consistent with list of all commands

yixiann committed:
Update url of ui image

Commit:
Modify gradle to enable application jar to run

Commit:
Merge branch 'master' of https://github.com/yixiann/ip

Commit:
Rename static variables to screaming snake case

Commit:
Apply SLAP to Parser, edit some comments and modify secondary command enum to have delimiter at end

Commit:
Fix typo and abstract out catchphrase

Commit:
Update Junit test

Commit:
Fix bug on initialising directory and rename delete handler

Commit:
Update README
